src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferDictionaryTryMethodsOverContainsKeyGuardAnalyzer.cs

```diff
@@ -591,6 +594,21 @@ private static bool IsAnySameReferenceOperation(IOperation source, ImmutableArra
             return false;
         }
 
+        private static IEnumerable<IOperation> GetDescendantNodesAndSelf(IOperation operation, Func<IOperation, bool> descendIntoChildren)
+        {
+            var childOperations = operation.Children.SelectMany(c =>
+            {
+                if (descendIntoChildren(c))
+                {
+                    return GetDescendantNodesAndSelf(c, descendIntoChildren);
+                }
+
+                return Enumerable.Empty<IOperation>();
+            });
+
+            return[operation, .. childOperations];
+        }
+
```

Comment:
Could use just inline conditional
```suggestion
        private static IEnumerable<IOperation> GetNonConditionalDescendantsAndSelf(IOperation operation)
        {
            var childOperations = operation.Children.SelectMany(c =>
            {
                if (c is not IConditionalOperation)
                {
                    return GetNonConditionalDescendantsAndSelf(c, descendIntoChildren);
                }

                return Enumerable.Empty<IOperation>();
            });

            return[operation, .. childOperations];
        }

```